esi_leap/common/flocx_market.py

```diff
@@ -0,0 +1,53 @@
+from esi_leap.db import api as dbapi
+from esi_leap.objects import flocx_market_client
+from esi_leap.objects import offer
+from keystoneauth1 import adapter
+from keystoneauth1 import loading as ks_loading
+from oslo_config import cfg
+import datetime
+
+CONF = cfg.CONF
+
+def retrieve_from_flocx_market():
+    auth_plugin = ks_loading.load_auth_from_conf_options(
+        CONF, 'flocx_market')
+    sess = ks_loading.load_session_from_conf_options(CONF, 'flocx_market',
+                                                         auth=auth_plugin)
+    adpt = adapter.Adapter(
+        session=sess,
+        service_type='marketplace',
+        interface='public')
+    marketplace_client = flocx_market_client.FlocxMarketClient(adpt)
```

Comment:
You shouldn't have to repeatedly re-initialize the client; instead, do something like this: https://github.com/CCI-MOC/esi-leap/blob/master/esi_leap/resource_objects/ironic_node.py

Reply:
fixed. I created a new function get_flocx_market_client().